inst/include/common/fims_vector.hpp

```diff
@@ -0,0 +1,483 @@
+#ifndef FIMS_VECTOR_HPP
+#define FIMS_VECTOR_HPP
+
+#include "../interface/interface.hpp"
+
+namespace  fims {
+
+
+/**
+ * Wrapper class for std::vector types. If this file is compiled with -DTMB_MODEL,
+ * conversion operators are defined for TMB vector types.
+ */
+template<typename Type>
+class Vector{
+    std::vector<Type> vec_m;
+    
+    /**
+     * @brief friiend comparison operator.
+     */
+    template<typename T>
+    friend bool operator==( const fims::Vector<T>& lhs,
+                           const fims::Vector<T>& rhs );
+public:
+    //Member Types
+    
+    typedef typename std::vector<Type>::value_type value_type; /*!<Member type Type>*/
+    typedef typename std::vector<Type>::allocator_type allocator_type; /*!<Allocator for type Type>*/
+    typedef typename std::vector<Type>::size_type size_type; /*!<Size type>*/
```

Comment:
I noticed the population.hpp loops we updated to use `size_t` instead of `int` for comparisons are now giving warnings again. Do we need to update the `size_type` of fims_vector somehow to match the size_t declaration for the loop iterators in population?

Reply:
I don't see this happening anymore on the branch - resolving